esp8266/modpybuart.c

```diff
@@ -70,16 +75,114 @@ STATIC void pyb_uart_init_helper(pyb_uart_obj_t *self, size_t n_args, const mp_o
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
 
+    // set baudrate
+    switch(args[ARG_baudrate].u_int) {
+        case 0:
+            break;
+        case 300:
+        case 600:
+        case 1200:
+        case 2400:
+        case 4800:
+        case 9600:
+        case 19200:
+        case 38400:
+        case 57600:
+        case 74880:
+        case 115200:
+        case 230400:
+        case 256000:
+        case 460800:
+        case 921600:
+        case 1843200:
+        case 3686400:
+            self->baudrate = args[ARG_baudrate].u_int;
+            UartDev.baut_rate = self->baudrate; // Sic!
```

Comment:
Some odd baud rates I've used/seen: 31250 for MIDI, 250,000, 500,000, and 1,000,000 for talking to bioloid servos.